### ISSUE TITLE
Swift Runtime Crash Under __swift_instantiateConcreteTypeFromMangledName in 15.4

### ISSUE THREAD

**Issue by alexdainiak** [Alamofire/Alamofire#3578]
## What did you do?

I used the PinnedCertificatesTrustEvaluator.

Before update to iOS 15.4.0 everething was good. But after the update, the application sometimes started to crash
Crashes for no reason (not every device, about 50 from 60000 devices), with iOS version 15.4.0.

## Alamofire Environment

**Alamofire Version:**5.5.0
**Dependency Manager:**SPM
**Xcode Version:**13.2.1
**Swift Version:**5.5
**Platform(s) Running Alamofire:**15.4.0
**macOS Version Running Xcode:**

## Crash report

[Crashed_org.alamofire.session.rootQueue.pl.txt](https://github.com/Alamofire/Alamofire/files/8284434/Crashed_org.alamofire.session.rootQueue.pl.txt)

> Crashed: org.alamofire.session.rootQueue
0  libsystem_kernel.dylib         0x6bbc __pthread_kill + 8
1  libsystem_pthread.dylib        0xd854 pthread_kill + 208
2  libsystem_c.dylib              0x1f6a4 abort + 124
3  libswiftCore.dylib             0x2ed01c swift::warningv(unsigned int, char const*, char*) + 138
4  libswiftCore.dylib             0x2f6698 swift::MetadataAllocator::Allocate(unsigned long, unsigned long) + 516
5  libswiftCore.dylib             0x2f71f4 _swift_getGenericMetadata(swift::MetadataRequest, void const* const*, swift::TargetTypeContextDescriptor<swift::InProcess> const*) + 1004
6  libswiftCore.dylib             0x2dad68 __swift_instantiateCanonicalPrespecializedGenericMetadata + 32
7  libswiftCore.dylib             0x318d10 (anonymous namespace)::DecodedMetadataBuilder::createBoundGenericType(swift::TargetContextDescriptor<swift::InProcess> const*, __swift::__runtime::llvm::ArrayRef<swift::TargetMetadata<swift::InProcess> const*>, swift::TargetMetadata<swift::InProcess> const*) const + 332
8  libswiftCore.dylib             0x3175a0 swift::Demangle::__runtime::TypeDecoder<(anonymous namespace)::DecodedMetadataBuilder>::decodeMangledType(swift::Demangle::__runtime::Node*, unsigned int, bool) + 19196
9  libswiftCore.dylib             0x312024 swift_getTypeByMangledNodeImpl(swift::MetadataRequest, swift::Demangle::__runtime::Demangler&, swift::Demangle::__runtime::Node*, void const* const*, std::__1::function<swift::TargetMetadata<swift::InProcess> const* (unsigned int, unsigned int)>, std::__1::function<swift::TargetWitnessTable<swift::InProcess> const* (swift::TargetMetadata<swift::InProcess> const*, unsigned int)>) + 552
10 libswiftCore.dylib             0x311d9c swift_getTypeByMangledNode + 540
11 libswiftCore.dylib             0x312510 swift_getTypeByMangledNameImpl(swift::MetadataRequest, __swift::__runtime::llvm::StringRef, void const* const*, std::__1::function<swift::TargetMetadata<swift::InProcess> const* (unsigned int, unsigned int)>, std::__1::function<swift::TargetWitnessTable<swift::InProcess> const* (swift::TargetMetadata<swift::InProcess> const*, unsigned int)>) + 968
12 libswiftCore.dylib             0x30fd14 swift_getTypeByMangledName + 540
13 libswiftCore.dylib             0x30ff28 swift_getTypeByMangledNameInContext + 132
14 Targetname                      0x844ac __swift_instantiateConcreteTypeFromMangledName + 4368319660 (<compiler-generated>:4368319660)
15 Targetname                      0x117d98 specialized _ContiguousArrayBuffer.init(_uninitializedCount:minimumCapacity:) + 4368924056
16 Targetname                      0x14dcfc specialized _ContiguousArrayBuffer._consumeAndCreateNew(bufferIsUnique:minimumCapacity:growForAppend:) + 4369145084
17 Targetname                      0xac854 specialized ContiguousArray._createNewBuffer(bufferIsUnique:minimumCapacity:growForAppend:) + 4368484436
18 Targetname                      0x125eac AlamofireExtension<A>.data.getter + 4368981676 (<compiler-generated>:4368981676)
19 Targetname                      0x125b5c PinnedCertificatesTrustEvaluator.evaluate(_:forHost:) + 616 (ServerTrustEvaluation.swift:616)
20 Targetname                      0x1261a0 protocol witness for ServerTrustEvaluating.evaluate(_:forHost:) in conformance PinnedCertificatesTrustEvaluator + 4368982432 (<compiler-generated>:4368982432)
21 Targetname                      0x13d024 SessionDelegate.attemptServerTrustAuthentication(with:) + 134 (SessionDelegate.swift:134)
22 Targetname                      0x13fa98 specialized SessionDelegate.urlSession(_:task:didReceive:completionHandler:) + 99 (SessionDelegate.swift:99)
23 Targetname                      0x13e7c4 @objc SessionDelegate.urlSession(_:task:didReceive:completionHandler:) + 4369082308
24 CFNetwork                      0x7d898 _CFHostIsDomainTopLevelForCertificatePolicy + 10480
25 Foundation                     0x4e26c __NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__ + 16
26 Foundation                     0x5f920 -[NSBlockOperation main] + 100
27 Foundation                     0x39a8c __NSOPERATION_IS_INVOKING_MAIN__ + 20
28 Foundation                     0x49f20 -[NSOperation start] + 764
29 Foundation                     0x4d368 __NSOPERATIONQUEUE_IS_STARTING_AN_OPERATION__ + 20
30 Foundation                     0x5ab7c __NSOQSchedule_f + 180
31 libdispatch.dylib              0x46048 _dispatch_block_async_invoke2 + 104
32 libdispatch.dylib              0x64094 _dispatch_client_callout + 16
33 libdispatch.dylib              0x3f4a4 _dispatch_lane_serial_drain$VARIANT$armv81 + 600
34 libdispatch.dylib              0x3ff78 _dispatch_lane_invoke$VARIANT$armv81 + 440
35 libdispatch.dylib              0x3f380 _dispatch_lane_serial_drain$VARIANT$armv81 + 308
36 libdispatch.dylib              0x3ff44 _dispatch_lane_invoke$VARIANT$armv81 + 388
37 libdispatch.dylib              0x498e0 _dispatch_workloop_worker_thread + 608
38 libsystem_pthread.dylib        0x1e10 _pthread_wqthread + 284
39 libsystem_pthread.dylib        0x193c start_wqthread + 8

**Comment by michaelversus** [comment]
I have the same exact issue

**Comment by brettlyon** [comment]
Also started seeing the same thing after bumping Alamofire from version `4.8.2` -> `5.5.0`. Unable to reproduce, nothing's coming up using the address or thread sanitizers, but it was the top crash in production until the version bump was reverted.

All crash reports occuring on devices running iOS `15.4.0`.

Alamofire Environment
**Alamofire Version:**5.5
**Dependency Manager:**Cocoapods
**Xcode Version:**13.2.1
**Swift Version:**5.5
**Platform(s) Running Alamofire:**15.4.0
macOS Version Running Xcode: 11.5.2

## Stack Trace
```
Crashed: org.alamofire.session.rootQueue
0  libsystem_kernel.dylib         0x6bbc __pthread_kill + 8
1  libsystem_pthread.dylib        0xd854 pthread_kill + 208
2  libsystem_c.dylib              0x1f6a4 abort + 124
3  libswiftCore.dylib             0x2ed01c swift::warningv(unsigned int, char const*, char*) + 138
4  libswiftCore.dylib             0x2f6698 swift::MetadataAllocator::Allocate(unsigned long, unsigned long) + 516
5  libswiftCore.dylib             0x2fd230 swift_getExistentialTypeMetadata + 1940
6  libswiftCore.dylib             0x31533c swift::Demangle::__runtime::TypeDecoder<(anonymous namespace)::DecodedMetadataBuilder>::decodeMangledType(swift::Demangle::__runtime::Node*, unsigned int, bool) + 10392
7  libswiftCore.dylib             0x312ee4 swift::Demangle::__runtime::TypeDecoder<(anonymous namespace)::DecodedMetadataBuilder>::decodeMangledType(swift::Demangle::__runtime::Node*, unsigned int, bool) + 1088
8  libswiftCore.dylib             0x312024 swift_getTypeByMangledNodeImpl(swift::MetadataRequest, swift::Demangle::__runtime::Demangler&, swift::Demangle::__runtime::Node*, void const* const*, std::__1::function<swift::TargetMetadata<swift::InProcess> const* (unsigned int, unsigned int)>, std::__1::function<swift::TargetWitnessTable<swift::InProcess> const* (swift::TargetMetadata<swift::InProcess> const*, unsigned int)>) + 552
9  libswiftCore.dylib             0x311d9c swift_getTypeByMangledNode + 540
10 libswiftCore.dylib             0x312510 swift_getTypeByMangledNameImpl(swift::MetadataRequest, __swift::__runtime::llvm::StringRef, void const* const*, std::__1::function<swift::TargetMetadata<swift::InProcess> const* (unsigned int, unsigned int)>, std::__1::function<swift::TargetWitnessTable<swift::InProcess> const* (swift::TargetMetadata<swift::InProcess> const*, unsigned int)>) + 968
11 libswiftCore.dylib             0x30fd14 swift_getTypeByMangledName + 540
12 libswiftCore.dylib             0x30ff28 swift_getTypeByMangledNameInContext + 132
13 brightwheel                    0x250200 __swift_instantiateConcreteTypeFromMangledName + 52 (<compiler-generated>:52)
14 brightwheel                    0x768be4 outlined destroy of EventMonitor? + 24
15 brightwheel                    0x765444 SessionDelegate.attemptServerTrustAuthentication(with:) + 131 (SessionDelegate.swift:131)
16 brightwheel                    0x767fa8 specialized SessionDelegate.urlSession(_:task:didReceive:completionHandler:) + 99 (SessionDelegate.swift:99)
17 brightwheel                    0x766cd4 @objc SessionDelegate.urlSession(_:task:didReceive:completionHandler:) + 132
18 CFNetwork                      0x7d898 _CFHostIsDomainTopLevelForCertificatePolicy + 10480
19 Foundation                     0x4e26c __NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__ + 16
20 Foundation                     0x5f920 -[NSBlockOperation main] + 100
21 Foundation                     0x39a8c __NSOPERATION_IS_INVOKING_MAIN__ + 20
22 Foundation                     0x49f20 -[NSOperation start] + 764
23 Foundation                     0x4d368 __NSOPERATIONQUEUE_IS_STARTING_AN_OPERATION__ + 20
24 Foundation                     0x5ab7c __NSOQSchedule_f + 180
25 libdispatch.dylib              0x46048 _dispatch_block_async_invoke2 + 104
26 libdispatch.dylib              0x64094 _dispatch_client_callout + 16
27 libdispatch.dylib              0x3f4a4 _dispatch_lane_serial_drain$VARIANT$armv81 + 600
28 libdispatch.dylib              0x3ff78 _dispatch_lane_invoke$VARIANT$armv81 + 440
29 libdispatch.dylib              0x3f380 _dispatch_lane_serial_drain$VARIANT$armv81 + 308
30 libdispatch.dylib              0x3ff44 _dispatch_lane_invoke$VARIANT$armv81 + 388
31 libdispatch.dylib              0x498e0 _dispatch_workloop_worker_thread + 608
32 libsystem_pthread.dylib        0x1e10 _pthread_wqthread + 284
33 libsystem_pthread.dylib        0x193c start_wqthread + 8
```

**Comment by mstfy** [comment]
We are experiencing same issue. It only happens on ios 15.4. We ship our app with xcode 13.2.1. Maybe xcode 13.3 would fix the issue.

**Comment by arctouch-rodrigoalmeida** [comment]
Facing the same issue on iOS 15.4

**Comment by Sky-Titan** [comment]
I have experienced same issue on iOS 15.4

**Comment by heinzl** [comment]
Same here!

**Comment by michaelversus** [comment]
<img width="1273" alt="Screenshot 2022-03-24 at 6 50 50 PM" src="https://user-images.githubusercontent.com/10232018/159968536-3b7fce7e-8d33-4139-bfac-69f025857238.png">

I think the issue is related with this one [here](https://developer.apple.com/forums/thread/697070)

Add following options to “Other Swift Flags” in the build settings:
`-Xllvm -sil-disable-pass=alloc-stack-hoisting`

or update to Xcode 13.3 might solve the problem

**Comment by StasKeiss** [comment]
We are having the exact same thing with iOS 15.4, Xcode 13.3. Around 500 users per week, still no solution?

**Comment by jshier** [comment]
This appears to be Swift runtime crash, so you should report it to Apple and post your report number here so others can duplicate it. There's not much Alamofire itself can do here. If you're seeing `EventMonitor` in the backtrace and you aren't using that feature you can effectively turn it off by passing an empty array to `Session` when initializing. `Session(..., eventMonitors: [])`, which should allow the compiler to optimize it away and perhaps avoid the crash.

**Comment by jshier** [comment]
This may be fixed in iOS 15.4.1 and the other OS updates that came out today (3/31). Please check to see whether crashes are still occurring on those versions. This bug may also be limited to devices before the iPhone X, so you can double check that as well.

**Comment by jshier** [comment]
I believe this is fixed in 15.4.1, 12.3.1, etc., so I'm closing the issue. Thanks for the reports everyone!

**Comment by alexdainiak** [comment]
> I believe this is fixed in 15.4.1, 12.3.1, etc., so I'm closing the issue. Thanks for the reports everyone!

Yes, indeed, only on 15.4.0 this issue is noticed. Users with 15.4.1 os version did not experience crashes. Thank you everybody!

**Comment by xianliy** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我星期六休息，您的信息我会在工作时间，尽快处理。Hello, I'm free on Saturday, your information in time I will work, as soon as possible to deal with.
